src/modules/75-cf/components/DetailPageTemplate/DetailPageTemplate.tsx

```diff
@@ -28,50 +28,58 @@ export interface DetailPageTemplateProps {
 
   /** Optional extra components to be added into header (context menu, edit button, etc...)  */
   /** Note: Use absolute position to style it */
-  headerExtras?: React.ReactNode
+  menuItems?: RbacOptionsMenuButtonProps['items']
+  metaData?: Record<string, string>
 }
 
 export const DetailPageTemplate: React.FC<DetailPageTemplateProps> = ({
   breadcrumbs,
   title,
   subTitle,
   identifier,
-  headerExtras,
-  children
+  children,
+  menuItems = [],
+  metaData
 }) => {
+  const { getString } = useStrings()
+
   return (
     <>
-      <Container
-        height={HEADER_HEIGHT}
-        padding={{ top: 'large', right: 'xlarge', bottom: 'large', left: 'xlarge' }}
-        style={{
-          background: 'rgba(219, 241, 255, .46)',
-          position: 'relative',
-          borderBottom: '.5px solid rgb(227 224 224 / 50%)'
-        }}
-      >
-        <Layout.Vertical spacing="medium">
-          <NGBreadcrumbs
-            customPathParams={{ module: 'cf' }}
-            links={breadcrumbs.map(({ title: label, url }) => ({ label, url }))}
-          />
-          <Container>
-            <Layout.Horizontal spacing="medium">
-              <Text style={{ fontSize: '20px', color: 'var(--black)' }}>{title}</Text>
-              {identifier && <IdentifierText identifier={identifier} allowCopy style={{ marginBottom: 0 }} />}
+      <Page.Header
+        size="large"
+        breadcrumbs={<NGBreadcrumbs customPathParams={{ module: 'cf' }} links={breadcrumbs} />}
+        title={
+          <>
+            <Layout.Horizontal spacing="small">
+              <Heading level={3} font={{ variation: FontVariation.H4 }}>
+                {title}
+              </Heading>
+              {identifier && <IdentifierText identifier={identifier} allowCopy />}
             </Layout.Horizontal>
             {subTitle && (
               <Text color={Color.GREY_400} padding={{ top: 'xsmall' }}>
                 {subTitle}
               </Text>
             )}
-          </Container>
-        </Layout.Vertical>
-        {headerExtras}
-      </Container>
-      <Container style={{ height: `calc(100% - ${HEADER_HEIGHT}px)`, overflow: 'auto', background: '#e4ebf433' }}>
-        {children}
-      </Container>
+          </>
+        }
+        content={
+          <>
+            {!!menuItems.length && (
+              <Container className={css.optionsMenu}>
+                <RbacOptionsMenuButton items={menuItems} />
+              </Container>
+            )}
+            {Object.entries(metaData as Record<string, string>).map(([key, val]) => (
+              <div className={css.metaData} key={key}>
+                <dt>{getString(key as keyof StringsMap)}</dt>
+                <dd>{val}</dd>
+              </div>
+            ))}
```

Comment:
Looks like `metaData` is actually marked as optional. Probably should check it actually exists first. Also, you don't have the `dl` tag and this will actually potentially put a bunch of meta data entries right on top of each other.
```suggestion
            {!!metaDataEntries.length && (
              <dl className={css.metaData}>
                {metaDataEntries.map(([key, val]) => (
                  <div key={key}>
                    <dt>{getString(key)}</dt>
                    <dd>{val}</dd>
                  </div>
                )}
              </dl>
            )}
```

And then above,
```typescript
const metaDataEntries = useMemo<[StringKeys, string]>(
  () => Object.entries(metaData || {}),
  [metaData]
)
```

Reply:
@knagurski IIRC Crystal had changed this code based a suggestion you made last week in this PR?

Reply:
@chrisblakely01 yep, but this does not match my suggested change and introduces a few bugs

Reply:
hm but why is this new suggestion adding more changes rather that just tweaking what I've changed?

Reply:
@knagurski actually yes it does match. Your suggestion is here https://github.com/harness/harness-core-ui/pull/8136#discussion_r844957954 and you didn't request a `dl` in that change, nor did you suggest any null checking.

You've just completely rewritten your own suggestion

Reply:
As you will see from the suggestion, I only suggested a change to the `dt` and `dd` part, not the surrounding tag. The change in the code moves the surrounding code into the loop, which will cause a bug. I did not mention the surround missing tag before as I did not spot it.